controllers/greptimedbcluster/deployers/datanode.go

```diff
@@ -153,6 +257,57 @@ func (d *DatanodeDeployer) deleteStorage(ctx context.Context, cluster *v1alpha1.
 	return nil
 }
 
+// isPodRestart checks if the pod needs to be restarted. For convenience, we only compare the necessary fields.
+// TODO(zyy17): Do we have a easy way to implement this?
+func (d *DatanodeDeployer) isPodRestart(new, old appsv1.StatefulSet) bool {
+	var (
+		newPodTemplate = new.Spec.Template
+		oldPodTemplate = old.Spec.Template
+	)
+
+	if !reflect.DeepEqual(newPodTemplate.GetObjectMeta().GetAnnotations(), oldPodTemplate.GetObjectMeta().GetAnnotations()) {
+		return true
+	}
+
+	if newPodTemplate.Spec.InitContainers[0].Image != oldPodTemplate.Spec.InitContainers[0].Image {
+		return true
+	}
+
+	// If the tolerations, affinity, nodeSelector are changed, the original Pod may need to be restarted for re-scheduling.
+	if !reflect.DeepEqual(newPodTemplate.Spec.Tolerations, oldPodTemplate.Spec.Tolerations) ||
+		!reflect.DeepEqual(newPodTemplate.Spec.Affinity, oldPodTemplate.Spec.Affinity) ||
+		!reflect.DeepEqual(newPodTemplate.Spec.NodeSelector, oldPodTemplate.Spec.NodeSelector) {
+		return true
+	}
+
+	// Compare the main container settings.
+	newMainContainer := newPodTemplate.Spec.Containers[constant.MainContainerIndex]
+	oldMainContainer := oldPodTemplate.Spec.Containers[constant.MainContainerIndex]
```

Comment:
question：By checking whether old.spec is equal to new.spec, we can determine whether to restart?

Reply:
We don't have to turn on maintenance mode when scaling. Maybe `isPodRestart()` is not a appropriate name, let me change to `isOldPodRestart()`.